lib/Sabberworm/CSS/CSSList/CSSList.php

```diff
@@ -110,15 +111,34 @@ private static function parseAtRule(ParserState $oParserState) {
 			$oLocation = URL::parse($oParserState);
 			$oParserState->consumeWhiteSpace();
 			$sMediaQuery = null;
-			if (!$oParserState->comes(';')) {
-				$sMediaQuery = $oParserState->consumeUntil(';');
+			try {
+				if (!$oParserState->comes(';')) {
+					$sMediaQuery = $oParserState->consumeUntil(';');
+				}
+				$oParserState->consume(';');
+			} catch (UnexpectedEOFException $e) {
+				// Save the media query if present
+				$sMediaQuery = '';
+				try {
+					while (!$oParserState->isEnd()) {
+						$sMediaQuery .= $oParserState->consume(1);
+					}
+				} catch (UnexpectedEOFException $e) {}
+
+				$sMediaQuery = trim($sMediaQuery);
```

Comment:
I prefer calling `consumeWhiteSpace` before and after reading the media query instead of trimming afterwards.

Reply:
Sorry… when using `consumeUntil(';')` this will of course include trailing whitespace if any so you do need `trim` after all…